aws-android-sdk-iot/src/main/java/com/amazonaws/mobileconnectors/iot/AWSIotMqttManager.java

```diff
@@ -210,6 +215,39 @@ public boolean isMetricsEnabled() {
     /** Override value for System.currentTimeInMillis.  Used for unit testing reconnect logic. */
     private Long unitTestMillisOverride;
 
+    /**
+     * Sets the userMetaData map.
+     *
+     * @param userMetaData userMetaData map
+     */
+    public void addUserMetaData(Map<String, String> userMetaData) {
+        this.userMetaData = userMetaData;
+
+        StringBuilder userMetadata = getUserMetaData();
+
+        if(userMetadata.length() > 225) {
```

Comment:
Best not to throw an IllegalArgumentException here - instead, log error and truncate the data so that it fits.

Reply:
Done